### PR TITLE
Daemon functions added for start,restart,stop,status and info from command line

### PR DESCRIPTION
**Pull request by amarendra-dianomic**

### CODE REVIEW

**Comment by Terris-Dianomic** [comment]
Lgtm

File: src/python/foglamp_daemon.py [hunk 1]
```diff
@@ -26,31 +48,121 @@ def do_something(logf):
     logger.addHandler(file_handler)
     logger.setLevel(logging.DEBUG)
```
Comment (praveen-garg) [inline src/python/foglamp_daemon.py]: level should be WARNING

File: src/python/foglamp_daemon.py [hunk 2]
```diff
@@ -26,31 +48,121 @@ def do_something(logf):
...
+def stop():
+    """
+    Stop the daemon
+    """
+
+    # Get the pid from the pidfile
+    pid = get_pid()
+
+    if pid is None:
+        message = "pidfile %s does not exist. Daemon not running?\n"
```
Comment (praveen-garg) [inline src/python/foglamp_daemon.py]: remove ? from the `running?`

File: src/python/foglamp_daemon.py [hunk 3]
```diff
@@ -26,31 +48,121 @@ def do_something(logf):
...
+    return is_running()
+
+
+def restart():
+    """
+    Relaunches the daemon
+
+    :param pidf: pidfile
+    :param logf: log file
+    :param wdir: working directory
```
Comment (praveen-garg) [inline src/python/foglamp_daemon.py]: param ?

File: src/python/foglamp_daemon.py [hunk 4]
```diff
@@ -26,31 +48,121 @@ def do_something(logf):
...
+    if is_running():
+        stop()
+    start()
+
+    return is_running()
+
+
+def is_running():
+    """
+    Check if the daemon is running.
```
Comment (praveen-garg) [inline src/python/foglamp_daemon.py]: what does it `:return ...` explain in docstring

File: src/python/foglamp_daemon.py [hunk 5]
```diff
@@ -26,31 +48,121 @@ def do_something(logf):
...
+    return get_pid() is not None
+
+
+def get_pid():
+    """
+    Returns the PID from pidf
+    """
+
+    try:
+        pf = open(pidf,'r')
```
Comment (praveen-garg) [inline src/python/foglamp_daemon.py]: should not we use context for reading file?

Comment (amarendra-dianomic) [inline src/python/foglamp_daemon.py]: Cannot. with statement does not catch exception. It is not allowed https://mail.python.org/pipermail//python-ideas/2011-November/012925.htmlhttps://mail.python.org/pipermail//python-ideas/2011-November/012925.html.

**Comment by amarendra-dianomic** [comment]
@praveen-garg all changes done per feedback. Pl check.

**Comment by praveen-garg** [comment]
- [ ] move foglamp_daemon.py to foglamp/server.py
- [ ] `if __name__ == __main__` should call `run`; 
- [ ] setup.py 

           entry_points={
               'console_scripts': [
                   'foglamp = foglamp.server:run',
                   'foglampd = foglamp.server:start',
               ],
            },

**Comment by Terris-Dianomic** [comment]
praveen, please change your comment to a change request

**Comment by amarendra-dianomic** [comment]
@Terris-Dianomic @praveen-garg all changes done as advised. Pl check.

**Pull request merged**